### DISCUSSION TITLE
Inertiajs Meta Tag preview is not working in third party site

### DISCUSSION THREAD

**Discussion by hlaporthein** [inertiajs/inertia#1113] [Help]
I am using Inertia js with Vue 3 and Laravel. Everthing is working fine except meta. I could view meta tag in source code on browser, meta tag is injected through inertia.js but when I share twitter or facebook, preview image or description is not working.

Below is the code I created custom meta component to reusable Meta.vue file
```
<template>
      <Head>
         <title>{{title}}</title>
         <meta charset="utf-8">
            <meta name="viewport" content="width=device-width, initial-scale=1">
         <link rel="shortcut icon" :href="favicon" type="image/x-icon">
         <link rel="icon" :href="favicon" type="image/x-icon">
         <meta name="title" :content="title">
         <meta name="description" :content="description">
         <meta property="og:type" content="website">
         <meta property="og:url" :content="link">
         <meta property="og:title" :content="title">
         <meta property="og:description" :content="description">
         <meta property="og:image" :content="image">
         <meta property="twitter:card" content="summary_large_image">
         <meta property="twitter:url" :content="link">
         <meta property="twitter:title" :content="title">
         <meta property="twitter:description" :content="description">
         <meta property="twitter:image" :content="image">       
         <slot />  
      </Head>
</template>

<script>
import { Head } from '@inertiajs/inertia-vue3'
import {SEO} from '@/Utilities/Constant';
import {seoImageUrl, faviconUrl} from '@/Utilities/CommonFunctions';

export default {
   components: {
      Head
   },
   props: {
      title: {
         type: String,
         default: SEO.title
      },
      description: {
         type: String,
         default: SEO.description
      },
      image: {
         type: String,
         default: seoImageUrl()
      },
      link: {
         type: String,
         default: SEO.link
      }
   },
   setup(props) {
      const favicon = faviconUrl();

      return {
         favicon
      }
   }
}
</script>
```

When  I insert Meta.vue file in pages in vue like below
```
<template>
   <Meta title="Home page" description="Home description" image="https://res.cloudinary.com/hlaporthein/image/upload/v1629887904/images/seo/hlaporthein_maikzu.png"  />
   <app>

      <div class="container-fluid">
              <h1>Hello World</h1>
      </div>


      
   </app>
</template>

<script>
import { Link } from '@inertiajs/inertia-vue3';
import App from '@/Layouts/App';
import { reactive, ref } from 'vue';
import Meta from '@/Components/Meta/Meta';


export default {
    components: {
       Meta,
      Link,
      App
    },

}
</script>

```
The above code is working fine, meta tag is available in browser source but when I share link in facebook or twitter... meta title, image and description is not showing. What am I missing? I setup as documentation said.  I also check sample [Ping CRM](https://demo.inertiajs.com/) , it is not working either. Please let me know how should I debug or inspect to work

**Comment by ajnsn** [comment]
https://github.com/inertiajs/inertia/issues/952#issuecomment-962676794

**Reply by adgower** [reply]
I have tried this same concept using SSR, and I can't get valet share https://cards-dev.twitter.com/validator to show my card? Any tips?

**Comment by sarozpradhan64** [comment]
having similar issue, used meta tags but facebook debugger cannot identify the inertia head meta tags.

**Comment by shaikhfoysal** [comment]
I have the same issue. Somebody told me if you can make ssr then it will work on vue inertia. But it is not working. Thanks

**Comment by kicker10BOG** [comment]
This is still an issue in at least Discord, Groupme, and Facebook. Can we expect this to ever be fixed?

**Reply by hlaporthein** [reply]
This is the issues I opened. I forgot to mention the solution. Use SSR for inertia @ajnsn . That would fixed.

**Reply by kicker10BOG** [reply]
I actually found another way. You can access the props in blade using `$page['props']` so I set the tags using blade and added `inertia` to each tag. But then they disappear when Inertia loads, so you also have also set them in Inertia. Here's an example of a tag in blade. 

```<meta inertia property="title" content="{{ (isset($page['props']['metaTitle']) ? $page['props']['metaTitle'] . ' | '  : '') . config('app.name', 'Site Name') }}" />```
